page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/utils/ExpectedConditionsUtils.java

```diff
@@ -123,101 +124,60 @@ public static void waitUntilElementGoneFromDom(WebElement webElement, long timeo
                 LOG.debug("There is no element {} in dom", webElement, e);
                 return;
             }
-            sleep(1);
+            sleep(PROPERTIES.getTimeout());
         }
         throw new NoSuchElementException("Timed out after " + timeout + " milliseconds waiting for web element '" + webElement.toString() + "' gone from DOM");
     }
 
     /**
-     * @param element a {@link org.openqa.selenium.WebElement} object.
+     * @param element a {@link WebElement} object.
      */
     public static void waitUntilElementGetInvisible(WebElement element) {
-        new WebDriverWait(PageFactory.getDriver(), PageFactory.getTimeOutInSeconds())
+        new WebDriverWait(PageContext.getCurrentPage().getDriver(), PROPERTIES.getTimeout())
                 .until(ExpectedConditions.not(ExpectedConditions.visibilityOf(element)));
     }
 
     /**
      * <p>
      * waitForElementGetEnabled.</p>
      *
-     * @param webElement a {@link org.openqa.selenium.WebElement} object.
-     * @param timeout    a long.
-     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException TODO
+     * @param webElement a {@link WebElement} object.
+     * @param timeout in milliseconds
+     * @throws WaitException TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)